src/main/java/net/pms/util/ImagesUtil.java

```diff
@@ -145,4 +161,335 @@ public static void parseImageByImaging(File file, DLNAMediaInfo media) throws Im
+	 * padding. Preserves aspect ratio. Format support is limited to
+	 * that of {@link ImageIO}.
+	 *
+	 * @param imageByteArray the source image in a supported format.
+	 * @param width the new width.
+	 * @param height the new height.
+	 * @param padToSize Whether padding should be used if source aspect doesn't
+	 *                  match target aspect.
+	 * @return The scaled, {@code null} if the source is {@code null} or the
+	 *         source image if the operation fails.
+	 */
+
+	public static byte[] scaleImage(byte[] imageByteArray, int width, int height, boolean padToSize) {
+		if (width == 0 || height == 0) {
+			return imageByteArray;
+		}
+		return scaleImage(imageByteArray, width, height, ScaleType.EXACT, ImageFormat.SOURCE, padToSize);
+	}
+
+	/**
+	 * Converts and scales an image in one operation. Scaling can be with or
+	 * without padding. Preserves aspect ratio. Format support is limited to
+	 * that of {@link ImageIO}.
+	 *
+	 * @param imageByteArray the source image in a supported format.
+	 * @param width the new width or 0 to disable scaling.
+	 * @param height the new height or 0 to disable scaling.
+	 * @param scaleType the {@link ScaleType} to use when scaling.
+	 * @param outputFormat the {@link ImageFormat} to generate or
+	 *                     {@link ImageFormat#SOURCE} to preserve source format.
+	 * @param padToSize Whether padding should be used if source aspect doesn't
+	 *                  match target aspect.
+	 * @return The scaled and/or converted image, {@code null} if the source is
+	 *         {@code null} or the source image if the operation fails.
+	 */
+	public static byte[] scaleImage(byte[] imageByteArray, int width, int height, ScaleType scaleType, ImageFormat outputFormat, boolean padToSize) {
+		if (imageByteArray == null) {
+			return null;
+		}
+		if (outputFormat == null) {
+			outputFormat = ImageFormat.SOURCE;
+		}
+
+		try {
+			ImageFormat inputFormat = null;
+			BufferedImage bufferedImage;
+			ImageReaderResult imageResult = CustomImageReader.read(new ByteArrayInputStream(imageByteArray));
+			bufferedImage = imageResult.bufferedImage;
+			inputFormat = imageResult.imageFormat;
+
+			if (bufferedImage == null || inputFormat == null) { // ImageIO doesn't support the image format
+				LOGGER.warn("Failed to resize image because the source format is unknown");
+				return imageByteArray;
+			}
+
+			if (outputFormat.equals(ImageFormat.SOURCE)) {
+				outputFormat = inputFormat;
+			}
+
+			ByteArrayOutputStream out = new ByteArrayOutputStream();
+			if (width == 0 ||
+				height == 0 ||
+				(
+					ScaleType.MAX.equals(scaleType) &&
+					bufferedImage.getWidth() <= width &&
+					bufferedImage.getHeight() <= height
+				)
+			) {
+				//No resize, just convert
+				if (inputFormat.equals(outputFormat)) {
+					// Nothing to do, just return source
+					return imageByteArray;
+				}
+				ImageIO.write(bufferedImage, outputFormat.toString(),out);
```

Comment:
@Nadahar what about to use Thumbnailator
			
				Thumbnails.of(bufferedImage)
					.size(bufferedImage.getWidth(), bufferedImage.getHeight())
					.outputFormat(outputFormat.toString())
					.outputQuality(1.0f)
					.toOutputStream(out);

 to fix the error "Invalid argument to native writeImage" instead of using ImageIO.write